Assets/HoloToolkit/Utilities/Scripts/Extensions/HapticsExtensions.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_WSA
+using UnityEngine.XR.WSA.Input;
+#if !UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using Windows.Devices.Haptics;
+using Windows.Perception;
+using Windows.UI.Input.Spatial;
+#endif
+#endif
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Extensions for the InteractionSource class to add haptics.
+    /// </summary>
+    public static class HapticsExtensions
+    {
+#if UNITY_WSA
+        public static void StartHaptics(this InteractionSource interactionSource, float intensity)
+        {
+#if !UNITY_EDITOR
+            UnityEngine.WSA.Application.InvokeOnUIThread(() =>
+            {
+                IReadOnlyList<SpatialInteractionSourceState> sources = SpatialInteractionManager.GetForCurrentView().GetDetectedSourcesAtTimestamp(PerceptionTimestampHelper.FromHistoricalTargetTime(DateTimeOffset.Now));
+
+                foreach (SpatialInteractionSourceState sourceState in sources)
+                {
+                    if (sourceState.Source.Id.Equals(interactionSource.id))
+                    {
+                        SimpleHapticsController simpleHapticsController = sourceState.Source.Controller.SimpleHapticsController;
+                        foreach (SimpleHapticsControllerFeedback hapticsFeedback in simpleHapticsController.SupportedFeedback)
+                        {
+                            if (hapticsFeedback.Waveform.Equals(0x1004))
+                            {
+                                simpleHapticsController.SendHapticFeedback(hapticsFeedback, intensity);
+                                return;
+                            }
+                        }
+                    }
+                }
+            }, true);
+#endif
+        }
+
+        public static void StartHaptics(this InteractionSource interactionSource, float intensity, float durationInSeconds)
```

Comment:
if the duration isn't specified what does the API do? Buzz forever? If so, perhaps duration in seconds can be defaulted to float.maxvalue so we don't have basically identical functions for basically identical code paths.

Reply:
ahh, I see, one has a 'forduration' tacked on.   So maybe float.maxvalue can be special cased to call the non-duration version of the function.

Reply:
Ahh yes, I was trying to think of a way to combine the two methods, but didn't think of using float.maxvalue. Thanks!